web/packages/plugins/shared/common/src/components/TreeView.tsx

```diff
@@ -117,72 +118,56 @@ const TreeNodeComponent = (props: {
 }
 
 export const TreeView = (props: {
+  index: TreeNode[]
   onSelect: (node: TreeNode) => void
   NodeWrapper?: React.FunctionComponent<NodeWrapperProps>
   NodeWrapperOnClick?: (node: TreeNode) => void
 }) => {
-  const { onSelect, NodeWrapper, NodeWrapperOnClick } = props
-  // @ts-ignore
-  const { token } = useContext(AuthContext)
-  const appConfig = useContext(ApplicationContext)
-  const tree = new Tree(token, appConfig.visibleDataSources)
-  const [index, setIndex] = useState<TreeNode[]>([])
-  const [loading, setLoading] = useState<boolean>(false)
+  const { index, onSelect, NodeWrapper, NodeWrapperOnClick } = props
+  // Use a per TreeView state to keep track of expanded nodes.
+  // This is so clicking in one tree will not affect other TreeViews
+  const [expandedNodes, setExpandedNodes] = useState<{ [k: string]: boolean }>(
+    {}
+  )
 
   useEffect(() => {
-    setLoading(true)
-    tree
-      .init()
-      // @ts-ignore
-      .then(() => setIndex([...tree]))
-      .finally(() => setLoading(false))
+    let expandedNodes: { [k: string]: boolean } = {}
+    index.forEach((node: TreeNode) => {
+      // Initialize expanded state where only top level DataSources are expanded
+      expandedNodes[node.nodeId] = node.isDataSource
+    })
+    setExpandedNodes(expandedNodes)
```

Comment:
It looks like this part should expand the datasources, but this is what I get when I refresh the page (nothing is expanded)

![image](https://user-images.githubusercontent.com/69512295/172378220-3ecd33da-7526-4cfa-bbbe-2ce3255cc390.png)

however, the tree in the EntityPicker for example seems to be expanded correctly

Reply:
Good catch. Fixed by waiting for the loading to be complete before rendering the tree